api/v1alpha1/healthcheck_types.go

```diff
@@ -25,11 +25,13 @@ import (
 // Important: Run "make" to regenerate code after modifying this file
 
 // HealthCheckSpec defines the desired state of HealthCheck
+// Either RepeatAfterSec or Scheduler must be defined for the health check to run
 type HealthCheckSpec struct {
-	RepeatAfterSec int      `json:"repeatAfterSec"`
-	Description    string   `json:"description,omitempty"`
-	Workflow       Workflow `json:"workflow"`
-	Level          string   `json:"level,omitempty"` // defines if a workflow runs in a Namespace or Cluster level
+	RepeatAfterSec int           `json:"repeatAfterSec,omitempty"`
+	Description    string        `json:"description,omitempty"`
+	Workflow       Workflow      `json:"workflow"`
+	Level          string        `json:"level,omitempty"`    // defines if a workflow runs in a Namespace or Cluster level
+	Scheduler      SchedulerSpec `json:"schedule,omitempty"` // Scheduler defines schedule rules to run HealthCheck
```

Comment:
since you changed the json/yaml representation to be `schedule`, it probably makes the most sense to change the struct property, as well... otherwise, I can image it getting confusing for a future developer/user looking at `schedule` in yaml and not immediately understanding they need to use `Scheduler` in golang.

let me know if you think it does make sense to retain the different spellings between the two representations, and if so, why
```suggestion
	Schedule      ScheduleSpec `json:"schedule,omitempty"` // Scheduler defines schedule rules to run HealthCheck
```